nbodykit/measurestats.py

```diff
@@ -460,24 +455,34 @@ def project_to_basis(comm, x3d, y3d, edges, los='z', poles=[], symmetric=True):
         xsum.flat += numpy.bincount(multi_index, weights=scratch.flat, minlength=xsum.size)
     
         # take the sum of weights
-        scratch[...] = 1.0
-        # count modes not in singular plane twice
-        if symmetric: scratch[:, nonsingular] = 2.
+        scratch[...] = 1.0 
+        if symmetric: scratch[:, nonsingular] = 2. # count modes not in singular plane twice
         Nsum.flat += numpy.bincount(multi_index, weights=scratch.flat, minlength=Nsum.size)
 
-        scratch[...] = y3d[row]
-
-        # the singular plane is down weighted by 0.5
-        if symmetric: scratch[:, nonsingular] *= 2.
-        
         # weight P(k,mu) and sum the weighted values
         for iell, ell in enumerate(poles_):
-            weighted_y3d = scratch * (2*ell + 1.) * legendre(ell)(mu)
-            ysum[iell,...].flat += numpy.bincount(multi_index, weights=weighted_y3d.flat, minlength=Nsum.size)
+            
+            weighted_y3d = legendre(ell)(mu) * y3d[row]
+
+            # add conjugate for this kx, ky, kz, corresponding to 
+            # the (-kx, -ky, -kz) --> need to make mu negative for conjugate
+            # below is same as legendre(ell)(-mu[:,nonsingular]) * (y3d[row][:,nonsingular]).conj()
+            if symmetric:
+                if ell % 2:
+                    weighted_y3d.real[:, nonsingular] = 0.
+                    weighted_y3d.imag[:, nonsingular] *= 2.
+                else:
+                    weighted_y3d.real[:, nonsingular] *= 2.
+                    weighted_y3d.imag[:, nonsingular] = 0.
+                    
+            weighted_y3d *= 0.5*(2*ell + 1.)
+            ysum[iell,...].real.flat += numpy.bincount(multi_index, weights=weighted_y3d.real.flat, minlength=Nsum.size)
+            if numpy.iscomplexobj(ysum):
+                ysum[iell,...].imag.flat += numpy.bincount(multi_index, weights=weighted_y3d.imag.flat, minlength=Nsum.size)
```

Comment:
I don't think I see what you mean. ysum is initialized on line 415 to be either real, or complex, depending on the type of the input y3d. And even if poles == [], we always compute ell = 0 in order to do P(k,mu)